Src/DataTypes/S3.cs

```diff
@@ -35,6 +35,19 @@ public S3(string serviceUrl, string bucket, string path, int batchSize) : base(D
             client = new AmazonS3Client(credentials, config);
         }
 
+        private Coord FromKey(string key)
+        {
+            string subPath = key.Remove(0, this.path.Length);
```

Comment:
what if the path is relative? (without .)
I think split have option to skip empty parts instead (if not linq can be used)

Reply:
This is a S3 key, the path cannot be relative.
It is retrieved from S3 itself and is not user input.